dbt/project.py

```diff
@@ -120,15 +122,18 @@ def compile_target(self, target_cfg):
             is_str = isinstance(value, dbt.compat.basestring)
 
             if is_str:
-                node = "config key: '{}'".format(key)
-                compiled_val = dbt.clients.jinja.get_rendered(value, ctx, node)
+                compiled_val = dbt.clients.jinja.get_rendered(value, ctx)
             else:
                 compiled_val = value
 
             compiled[key] = compiled_val
 
         return compiled
 
+    def compile_and_update_target(self):
+        target = self.cfg['target']
+        self.cfg['outputs'][target].update(self.run_environment())
```

Comment:
@cmcarthur i don't like playing with the state here, but I don't want to touch the `Project` interface. lmk if you agree